server/app/mutations/grid_services/helpers.rb

```diff
@@ -42,5 +42,17 @@ def update_grid_service(grid_service, force: false)
 
       save_grid_service(grid_service)
     end
+
+    # @param grid_service [GridService]
+    # @param state [String] desired state
+    # @param force [Boolean] force-update revision
+    def deploy_grid_service(grid_service, state = 'running', force: false)
+      grid_service.deploy_requested_at = Time.now.utc
+      grid_service.state = state
+
+      if force ? update_grid_service(grid_service, force: true) : save_grid_service(grid_service)
```

Comment:
A bit hard to read.

Reply:
This is existing code that was just moved from the `GridServices::Deploy#execute`. The control flow is for dealing with the non-exceptions-based mutations `add_error` case: do not create the deploy if the update/save fails.

Cleaning this up would probably be having the `update_grid_service`/`save_grid_service` return an error instead, but then it would get messy figuring out where to deal with those exceptions and convert them into mutations `add_error` instead.